src/settings/nm-settings.c

```diff
@@ -721,21 +716,19 @@ load_plugins (NMSettings *self, const char **plugins, GError **error)
 			continue;
 		}
 
-		if (find_plugin (list, pname))
+		full_name = g_strdup_printf ("nm-settings-plugin-%s", pname);
+		path = g_module_build_path (NMPLUGINDIR, full_name);
+
+		if (plugin_loaded (list, path))
 			continue;
 
 load_plugin:
```

Comment:
this label should be moved before
```
full_name = g_strdup_printf ("nm-settings-plugin-%s", pname);
```
Also, make sure that we don't leak `full_name` nor `path`. That means, use `nm_clear_g_free()`, either immediately after `load_plugin:` label, or immediately before `goto load_plugin;`.

If you dislike that this is handled via goto, you can also refactor the code to solve it somehow differently. But I doubt that would be much nicer...